Lib/socket.py

```diff
@@ -348,6 +349,9 @@ def makefile(self, mode="r", buffering=None, *,
     if hasattr(os, 'sendfile'):
 
         def _sendfile_use_sendfile(self, file, offset=0, count=None):
+            # Lazy import to improve module import time
+            import selectors
```

Comment:
I suggest to use a global variable to avoid the import at each call:

```
global selectors
if selectors is None:
    import selectors
```

You can define the selectors variable to None at the top of the file with a comment:

```
# module imported lazily
selectors = None
```

Reply:
I checked several recent PRs making module loading lazy, but most of them do not use the pattern with a global variable (I suspect for readability reasons). 

@vstinner Are there any other reasons besides the small performance improvement for using the global variable?

Reply:
@serhiy-storchaka: Do you think that it's still useful in 2024 to use a global variable to avoid `import selectors` at each function call?

Reply:
I'd just do the import. It's a mere dict lookup without a conditional when the import has already happened.